From: Rod Whitby <rod@whitby.id.au>
Subject: Enable support for up to 256MB on all ixp4xx targets.

Enabled a startup prefix command which initialises the memory
controller and scans for up to 256MB of RAM.

Signed-off-by: Rod Whitby <rod@whitby.id.au>
--

Index: apex/src/mach-ixp42x/debian-dsmg600-arm_config
===================================================================
--- apex.orig/src/mach-ixp42x/debian-dsmg600-arm_config	2008-05-28 10:52:52.000000000 +0930
+++ apex/src/mach-ixp42x/debian-dsmg600-arm_config	2008-05-28 10:53:13.000000000 +0930
@@ -140,7 +140,8 @@
 # CONFIG_ENV_REGION_KERNEL_SWAP is not set
 CONFIG_ENV_STARTUP_RAMDISK_COPY=y
 # CONFIG_ENV_REGION_RAMDISK_SWAP is not set
-# CONFIG_ENV_STARTUP_PREFIX_P is not set
+CONFIG_ENV_STARTUP_PREFIX_P=y
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
Index: apex/src/mach-ixp42x/debian-dsmg600-armeb_config
===================================================================
--- apex.orig/src/mach-ixp42x/debian-dsmg600-armeb_config	2008-05-28 10:52:52.000000000 +0930
+++ apex/src/mach-ixp42x/debian-dsmg600-armeb_config	2008-05-28 10:53:13.000000000 +0930
@@ -140,7 +140,8 @@
 # CONFIG_ENV_REGION_KERNEL_SWAP is not set
 CONFIG_ENV_STARTUP_RAMDISK_COPY=y
 # CONFIG_ENV_REGION_RAMDISK_SWAP is not set
-# CONFIG_ENV_STARTUP_PREFIX_P is not set
+CONFIG_ENV_STARTUP_PREFIX_P=y
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
Index: apex/src/mach-ixp42x/debian-nas100d-arm_config
===================================================================
--- apex.orig/src/mach-ixp42x/debian-nas100d-arm_config	2008-05-28 10:52:56.000000000 +0930
+++ apex/src/mach-ixp42x/debian-nas100d-arm_config	2008-05-28 10:53:13.000000000 +0930
@@ -137,7 +137,8 @@
 # CONFIG_ENV_REGION_KERNEL_SWAP is not set
 CONFIG_ENV_STARTUP_RAMDISK_COPY=y
 # CONFIG_ENV_REGION_RAMDISK_SWAP is not set
-# CONFIG_ENV_STARTUP_PREFIX_P is not set
+CONFIG_ENV_STARTUP_PREFIX_P=y
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
Index: apex/src/mach-ixp42x/debian-nas100d-armeb_config
===================================================================
--- apex.orig/src/mach-ixp42x/debian-nas100d-armeb_config	2008-05-28 10:52:56.000000000 +0930
+++ apex/src/mach-ixp42x/debian-nas100d-armeb_config	2008-05-28 10:53:13.000000000 +0930
@@ -137,7 +137,8 @@
 # CONFIG_ENV_REGION_KERNEL_SWAP is not set
 CONFIG_ENV_STARTUP_RAMDISK_COPY=y
 # CONFIG_ENV_REGION_RAMDISK_SWAP is not set
-# CONFIG_ENV_STARTUP_PREFIX_P is not set
+CONFIG_ENV_STARTUP_PREFIX_P=y
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
Index: apex/src/mach-ixp42x/debian-nslu2-16mb-arm_config
===================================================================
--- apex.orig/src/mach-ixp42x/debian-nslu2-16mb-arm_config	2008-05-28 10:52:50.000000000 +0930
+++ apex/src/mach-ixp42x/debian-nslu2-16mb-arm_config	2008-05-28 10:53:13.000000000 +0930
@@ -141,7 +141,7 @@
 CONFIG_ENV_STARTUP_RAMDISK_COPY=y
 CONFIG_ENV_REGION_RAMDISK_SWAP=y
 CONFIG_ENV_STARTUP_PREFIX_P=y
-CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+64m"
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
Index: apex/src/mach-ixp42x/debian-nslu2-16mb-armeb_config
===================================================================
--- apex.orig/src/mach-ixp42x/debian-nslu2-16mb-armeb_config	2008-05-28 10:52:50.000000000 +0930
+++ apex/src/mach-ixp42x/debian-nslu2-16mb-armeb_config	2008-05-28 10:53:13.000000000 +0930
@@ -141,7 +141,7 @@
 CONFIG_ENV_STARTUP_RAMDISK_COPY=y
 # CONFIG_ENV_REGION_RAMDISK_SWAP is not set
 CONFIG_ENV_STARTUP_PREFIX_P=y
-CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+64m"
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
Index: apex/src/mach-ixp42x/debian-nslu2-armeb_config
===================================================================
--- apex.orig/src/mach-ixp42x/debian-nslu2-armeb_config	2008-05-28 10:53:02.000000000 +0930
+++ apex/src/mach-ixp42x/debian-nslu2-armeb_config	2008-05-28 10:53:13.000000000 +0930
@@ -140,7 +140,8 @@
 # CONFIG_ENV_REGION_KERNEL_SWAP is not set
 CONFIG_ENV_STARTUP_RAMDISK_COPY=y
 # CONFIG_ENV_REGION_RAMDISK_SWAP is not set
-# CONFIG_ENV_STARTUP_PREFIX_P is not set
+CONFIG_ENV_STARTUP_PREFIX_P=y
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
Index: apex/src/mach-ixp42x/slugos-dsmg600-arm_config
===================================================================
--- apex.orig/src/mach-ixp42x/slugos-dsmg600-arm_config	2008-05-28 10:52:58.000000000 +0930
+++ apex/src/mach-ixp42x/slugos-dsmg600-arm_config	2008-05-28 10:53:13.000000000 +0930
@@ -133,7 +133,8 @@
 CONFIG_AUTOBOOT_DELAY=10
 CONFIG_ENV_STARTUP_KERNEL_COPY=y
 # CONFIG_ENV_REGION_KERNEL_SWAP is not set
-# CONFIG_ENV_STARTUP_PREFIX_P is not set
+CONFIG_ENV_STARTUP_PREFIX_P=y
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
Index: apex/src/mach-ixp42x/slugos-dsmg600-armeb_config
===================================================================
--- apex.orig/src/mach-ixp42x/slugos-dsmg600-armeb_config	2008-05-28 10:52:50.000000000 +0930
+++ apex/src/mach-ixp42x/slugos-dsmg600-armeb_config	2008-05-28 10:53:13.000000000 +0930
@@ -133,7 +133,8 @@
 CONFIG_AUTOBOOT_DELAY=10
 CONFIG_ENV_STARTUP_KERNEL_COPY=y
 # CONFIG_ENV_REGION_KERNEL_SWAP is not set
-# CONFIG_ENV_STARTUP_PREFIX_P is not set
+CONFIG_ENV_STARTUP_PREFIX_P=y
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
Index: apex/src/mach-ixp42x/slugos-fsg3-arm_config
===================================================================
--- apex.orig/src/mach-ixp42x/slugos-fsg3-arm_config	2008-05-28 10:53:11.000000000 +0930
+++ apex/src/mach-ixp42x/slugos-fsg3-arm_config	2008-05-28 10:53:13.000000000 +0930
@@ -134,7 +134,8 @@
 CONFIG_AUTOBOOT_DELAY=10
 CONFIG_ENV_STARTUP_KERNEL_COPY=y
 # CONFIG_ENV_REGION_KERNEL_SWAP is not set
-# CONFIG_ENV_STARTUP_PREFIX_P is not set
+CONFIG_ENV_STARTUP_PREFIX_P=y
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
Index: apex/src/mach-ixp42x/slugos-fsg3-armeb_config
===================================================================
--- apex.orig/src/mach-ixp42x/slugos-fsg3-armeb_config	2008-05-28 10:53:11.000000000 +0930
+++ apex/src/mach-ixp42x/slugos-fsg3-armeb_config	2008-05-28 10:53:13.000000000 +0930
@@ -134,7 +134,8 @@
 CONFIG_AUTOBOOT_DELAY=10
 CONFIG_ENV_STARTUP_KERNEL_COPY=y
 # CONFIG_ENV_REGION_KERNEL_SWAP is not set
-# CONFIG_ENV_STARTUP_PREFIX_P is not set
+CONFIG_ENV_STARTUP_PREFIX_P=y
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
Index: apex/src/mach-ixp42x/slugos-nas100d-arm_config
===================================================================
--- apex.orig/src/mach-ixp42x/slugos-nas100d-arm_config	2008-05-28 10:52:58.000000000 +0930
+++ apex/src/mach-ixp42x/slugos-nas100d-arm_config	2008-05-28 10:53:13.000000000 +0930
@@ -133,7 +133,8 @@
 CONFIG_AUTOBOOT_DELAY=10
 CONFIG_ENV_STARTUP_KERNEL_COPY=y
 # CONFIG_ENV_REGION_KERNEL_SWAP is not set
-# CONFIG_ENV_STARTUP_PREFIX_P is not set
+CONFIG_ENV_STARTUP_PREFIX_P=y
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
Index: apex/src/mach-ixp42x/slugos-nas100d-armeb_config
===================================================================
--- apex.orig/src/mach-ixp42x/slugos-nas100d-armeb_config	2008-05-28 10:52:56.000000000 +0930
+++ apex/src/mach-ixp42x/slugos-nas100d-armeb_config	2008-05-28 10:53:13.000000000 +0930
@@ -133,7 +133,8 @@
 CONFIG_AUTOBOOT_DELAY=10
 CONFIG_ENV_STARTUP_KERNEL_COPY=y
 # CONFIG_ENV_REGION_KERNEL_SWAP is not set
-# CONFIG_ENV_STARTUP_PREFIX_P is not set
+CONFIG_ENV_STARTUP_PREFIX_P=y
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
Index: apex/src/mach-ixp42x/slugos-nslu2-16mb-arm_config
===================================================================
--- apex.orig/src/mach-ixp42x/slugos-nslu2-16mb-arm_config	2008-05-28 10:53:06.000000000 +0930
+++ apex/src/mach-ixp42x/slugos-nslu2-16mb-arm_config	2008-05-28 10:53:13.000000000 +0930
@@ -136,7 +136,8 @@
 CONFIG_AUTOBOOT_DELAY=10
 CONFIG_ENV_STARTUP_KERNEL_COPY=y
 # CONFIG_ENV_REGION_KERNEL_SWAP is not set
-# CONFIG_ENV_STARTUP_PREFIX_P is not set
+CONFIG_ENV_STARTUP_PREFIX_P=y
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
Index: apex/src/mach-ixp42x/slugos-nslu2-16mb-armeb_config
===================================================================
--- apex.orig/src/mach-ixp42x/slugos-nslu2-16mb-armeb_config	2008-05-28 10:53:06.000000000 +0930
+++ apex/src/mach-ixp42x/slugos-nslu2-16mb-armeb_config	2008-05-28 10:53:13.000000000 +0930
@@ -136,7 +136,8 @@
 CONFIG_AUTOBOOT_DELAY=10
 CONFIG_ENV_STARTUP_KERNEL_COPY=y
 # CONFIG_ENV_REGION_KERNEL_SWAP is not set
-# CONFIG_ENV_STARTUP_PREFIX_P is not set
+CONFIG_ENV_STARTUP_PREFIX_P=y
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
Index: apex/src/mach-ixp42x/slugos-nslu2-arm_config
===================================================================
--- apex.orig/src/mach-ixp42x/slugos-nslu2-arm_config	2008-05-28 10:53:06.000000000 +0930
+++ apex/src/mach-ixp42x/slugos-nslu2-arm_config	2008-05-28 10:53:13.000000000 +0930
@@ -136,7 +136,8 @@
 CONFIG_AUTOBOOT_DELAY=10
 CONFIG_ENV_STARTUP_KERNEL_COPY=y
 # CONFIG_ENV_REGION_KERNEL_SWAP is not set
-# CONFIG_ENV_STARTUP_PREFIX_P is not set
+CONFIG_ENV_STARTUP_PREFIX_P=y
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
Index: apex/src/mach-ixp42x/slugos-nslu2-armeb_config
===================================================================
--- apex.orig/src/mach-ixp42x/slugos-nslu2-armeb_config	2008-05-28 10:53:06.000000000 +0930
+++ apex/src/mach-ixp42x/slugos-nslu2-armeb_config	2008-05-28 10:53:13.000000000 +0930
@@ -136,7 +136,8 @@
 CONFIG_AUTOBOOT_DELAY=10
 CONFIG_ENV_STARTUP_KERNEL_COPY=y
 # CONFIG_ENV_REGION_KERNEL_SWAP is not set
-# CONFIG_ENV_STARTUP_PREFIX_P is not set
+CONFIG_ENV_STARTUP_PREFIX_P=y
+CONFIG_ENV_STARTUP_PREFIX="sdram-init; memscan -u 0+256m"
 
 #
 #    Regions
